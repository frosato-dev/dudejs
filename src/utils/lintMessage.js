import load from "@commitlint/load/lib/index";
import lint from "@commitlint/lint/lib/index";

export default async commitMessage => {
  const CONFIG = require("./../configs/commitlint");

  const options = await load(CONFIG);
  const report = await lint(commitMessage, options.rules);

  return report.errors;
};
