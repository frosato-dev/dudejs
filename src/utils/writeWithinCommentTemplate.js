export default commands => `
#### DudeJS - START ####
${commands.join("\n")}
#### DudeJS - END ####
`;
