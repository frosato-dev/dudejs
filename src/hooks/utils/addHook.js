import path from "path";
import fs from "fs";
import readPkgUp from "read-pkg-up";
import writeWithinCommentTemplate from "./../../utils/writeWithinCommentTemplate";

const modulePath = fs.realpathSync(process.cwd());
const clientProjectPath = path.join(modulePath, "../..");
const gitHookPath = path.join(clientProjectPath, ".git", "hooks");

export default (hookName, commands = []) => {
  const { pkg: packageJson } = readPkgUp.sync({ cwd: clientProjectPath });

  if (!packageJson) {
    //eslint-disable-next-line
    console.log(
      `‚ùå DudeJS was not able to add ${hookName} hook, come on dude did you forget to git init?`,
    );
    process.exit();
  }

  const filePath = path.join(gitHookPath, hookName);

  if (fs.existsSync(filePath)) {
    //eslint-disable-next-line
    console.log(
      `üòé Seems you already have a ${hookName} hook, DudeJS hasn't done anything and he likes it.`,
    );
    process.exit();
  }

  const fileContent = writeWithinCommentTemplate(commands);
  fs.writeFileSync(filePath, fileContent, { mode: 0o755 });

  //eslint-disable-next-line
  console.log(`‚úÖ DudeJS successfully installed ${hookName} hook for you.`);
};
