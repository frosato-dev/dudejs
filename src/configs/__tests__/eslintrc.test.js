const mockHasDependency = jest.fn();

jest.mock("../../bin/utils/hasDependency", () => mockHasDependency);
jest.mock("../prettierrc", () => ({
  inheritedPrettierConfig: true,
}));

beforeEach(() => {
  jest.resetModules();
});

it("returns expected eslint configuration", () => {
  require("../../bin/utils/hasDependency").mockImplementation(() => false);

  expect(require("../eslintrc")).toMatchSnapshot();
});

it("returns expected eslint configuration when react found", () => {
  require("../../bin/utils/hasDependency").mockImplementation(name => name === "react");

  expect(require("../eslintrc")).toMatchSnapshot();
});

it("returns expected eslint configuration when jest found", () => {
  require("../../bin/utils/hasDependency").mockImplementation(name => name === "jest");

  expect(require("../eslintrc")).toMatchSnapshot();
});
