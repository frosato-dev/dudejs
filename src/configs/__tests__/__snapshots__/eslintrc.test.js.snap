// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns expected eslint configuration 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": Object {
      "global": true,
    },
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "prettier",
  ],
  "parser": "babel-eslint",
  "parserOptions": Object {
    "ecmaVersion": 2017,
    "sourceType": "module",
  },
  "plugins": Array [
    "prettier",
  ],
  "rules": Object {
    "prettier/prettier": Array [
      "error",
      Object {
        "inheritedPrettierConfig": true,
      },
    ],
  },
}
`;

exports[`returns expected eslint configuration when jest found 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": Object {
      "global": true,
    },
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "prettier",
    "plugin:jest/recommended",
  ],
  "parser": "babel-eslint",
  "parserOptions": Object {
    "ecmaVersion": 2017,
    "sourceType": "module",
  },
  "plugins": Array [
    "prettier",
    "jest",
  ],
  "rules": Object {
    "prettier/prettier": Array [
      "error",
      Object {
        "inheritedPrettierConfig": true,
      },
    ],
  },
}
`;

exports[`returns expected eslint configuration when react found 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": Object {
      "global": true,
    },
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "prettier",
    "plugin:react/recommended",
  ],
  "parser": "babel-eslint",
  "parserOptions": Object {
    "ecmaVersion": 2017,
    "sourceType": "module",
  },
  "plugins": Array [
    "prettier",
    "react",
  ],
  "rules": Object {
    "prettier/prettier": Array [
      "error",
      Object {
        "inheritedPrettierConfig": true,
      },
    ],
  },
}
`;
